torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -916,18 +927,37 @@ BufHandle TensorExprKernel::bindSymbolicShapeInput(
 
 Tensor TensorExprKernel::bindInput(const torch::jit::Value* input) {
   auto const& t = input->type();
+  auto const& outputs = input->owningGraph()->outputs();
+  std::unordered_set<const Value*> outputs_set(outputs.begin(), outputs.end());
   Tensor result(nullptr, nullptr);
   switch (t->kind()) {
     case TypeKind::TensorType: {
       auto tt = input->type()->cast<TensorType>();
       if (!input->isCompleteTensor()) {
         auto bufHandle =
             bindSymbolicShapeInput(input, "t" + input_name_map_[input]);
-        bufs_.emplace(input, bufHandle.node());
+
+        if (!outputs_set.count(input)) {
+          bufs_.emplace(input, bufHandle.node());
+        } else {
+          // If 'input' is also a graph output, then we need to produce a copy
+          std::vector<DimArg> inputTensorDims;
+          for (size_t i = 0; i < bufHandle.dims().size(); i++) {
+            inputTensorDims.emplace_back(
+                DimArg(bufHandle.dims()[i], "i" + c10::to_string(i)));
+          }
+          result = Compute(
```

Comment:
Can we try to combine the copying here with the copying below? The one below includes handling strides, but that can probably be an argument and the logic of copying would be in one place then.

Reply:
Let me see what we can do there!

Reply:
Please take a look at #70535! I extracted the changes to a separate PR as you suggested below and refactored the logic a little.